doc/theoretical_description_conformity_scores.rst

```diff
@@ -0,0 +1,93 @@
+scores when calculating the interval's bounds. If a score is symmetrical two
+quantiles will be computed : one on the right side of the distribution
+and the other on the left side.
+
+1. The absolute residual score
+==============================
+
+The absolute residual score (:class:`mapie.conformity_scores.AbsoluteResidualScore`)
+is the simplest and most commonly used conformal score, it translates the error
+of the model : in regression it is called the residual.
+
+.. math:: |Y-\hat{\mu}(X)|
+
+With this score the intervals of predictions will be constant over the whole dataset.
+This score is symmetric (*see above for definition*).
+
+2. The gamma score
+==================
+
+The gamma score (:class:`mapie.conformity_scores.GammaConformityScore`) adds a
+notion of adaptivity with the normalization of the residuals by the predictions.
+
+.. math:: \frac{|Y-\hat{\mu}(X)|}{\hat{\mu}(X)}
+
+It computes adaptive intervals : intervals of different size on each example.
+This score is asymmetric (*see definition above*).
+
+Compared to the absolute residual score, it allows to see regions with smaller intervals
+than others which are interpreted as regions with more certainty than others.
+
+3. The conformal residual fitting score
+=======================================
+
+The conformal residual fitting score (:class:`mapie.conformity_scores.ConformalizedResidualFittingScore`)
+(CRF) is slightly more complex than the previous scores.
+The normalization of the residual is now done by the predictions of an additional model
+:math:`\sigma` which learns to predict the base model residuals from :math:`X`.
+:math:`\sigma` is trained on :math:`(X, |Y-\hat{\mu}(X)|)` and the formula of th score is :
+
+.. math:: \frac{|Y-\hat{\mu}(X)|}{\hat{\sigma}(X)}
+
+This score provides adaptive intervals : intervals of different sizes in each point
+and is symmetric (*see definition above*). Unlike the scores above, and due to
+the additionnal model required this score can only be used with split methods.
+
+Normalisation by the learned residuals from :math:`X` adds to the score a knowledge of
+:math:`X` and its similarity to the other examples in the dataset. In fact, using this
+score results in even more adaptive intervals. Compared to the gamma score, the other adaptive
+score implemented in MAPIE, it maintains relevant interval sizes over the entire dataset
+even when there are outliers that could perturb the model. With gamma score, if
+there are strong outliers, the intervals over the hole datset are sometimes too large
+to be useful.
+Therefore, the interpretation of the intervals provided by the conformal residual fitting
+score can help to detect outliers.
+
+
+Key takeaways
+=============
+
+- The absolute residual score is the basic conformity score and gives constant intervals.
+- The gamma conformity score adds a notion of adaptivity by giving intervals of different sizes,
+  but it can give absurd results with strong outliers.
```

Comment:
```suggestion
  but it can give absurd results if the prediction model is too uncertain.
```

Reply:
I'm not sure to understand what you mean by "(if the uncertainty is of the order of magnitude of the predictions)"